special-pages/pages/new-tab/app/freemium-pir-banner/components/FreemiumPIRBanner.js

```diff
@@ -0,0 +1,56 @@
+import cn from 'classnames';
+import { h } from 'preact';
+import { Button } from '../../../../../shared/components/Button/Button';
+import { DismissButton } from '../../components/DismissButton';
+import styles from './FreemiumPIRBanner.module.css';
+import { FreemiumPIRBannerContext } from '../FreemiumPIRBannerProvider';
+import { useContext } from 'preact/hooks';
+
+/**
+ * @typedef { import("../../../types/new-tab").FreemiumPIRBannerMessage} FreemiumPIRBannerMessage
+ * @param {object} props
+ * @param {FreemiumPIRBannerMessage} props.message
+ * @param {(id: string) => void} props.dismiss
+ * @param {(id: string) => void} props.action
+ */
+
+export function FreemiumPIRBanner({ message, action, dismiss }) {
+    const convertMarkdownToHTML = (markdown) => {
+        // Use a regular expression to find all the words wrapped in **
+        const regex = /\*\*(.*?)\*\*/g;
+
+        // Replace the matched text with the HTML <strong> tags
+        const result = markdown.replace(regex, '<strong>$1</strong>');
+        return result;
+    };
```

Comment:
Can you pull this out into a separate file, and then we can write some unit tests against it. Anything we’re going put into `dangerouslySetInnerHTML` should have plenty of test cases to validate that we don’t break on things like XSS

Reply:
There are some example tests here you can use as reference https://github.com/duckduckgo/content-scope-scripts/blob/main/special-pages/unit-test/translations.mjs#L1-L57